builtin/logical/aws/secret_access_keys.go

```diff
@@ -183,6 +183,7 @@ func (b *backend) secretAccessKeysCreate(
 	// Create the user
 	_, err = iamClient.CreateUser(&iam.CreateUserInput{
 		UserName: aws.String(username),
+		Path:     aws.String(role.UserPath),
```

Comment:
This will produce an error if `role.UserPath` is an empty string. While the code you contributed will ensure that new roles will never have this be an empty string, that behavior isn't quite right, and roles that existed prior to this being implemented will also have `UserPath` be an empty string. You can either provide a role upgrade path to ensure that the `UserPath` is never an empty string when the `credential_type` is `iam_user`, or you can just put some code in here to only set `Path` when `UserPath` is not an empty string. I think the latter is probably better.

Reply:
Yeah true it will cause problems for users who are still using old configuration. What about third approach and set "Path" option to `/` if it is empty?

Reply:
Functionally the two approaches are identical, so I'd recommend whichever one leads to simpler code :)